python/pyspark/sql/dataframe.py

```diff
@@ -2153,6 +2153,22 @@ def transform(self, func):
                                               "should have been DataFrame." % type(result)
         return result
 
+    @since(3.0)
+    def sameSemantics(self, other):
+        """
+        Return true when the query plan of the given :class:`DataFrame` will return the same
+        results as this :class:`DataFrame`.
+        """
+        return self._jdf.sameSemantics(other)
```

Comment:
should be `self._jdf.sameSemantics(other._jdf)`

Reply:
It becomes the same error as the second one:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/liang.zhang/work/repos/apache/spark/python/pyspark/sql/dataframe.py", line 2162, in sameSemantics
    return self._jdf.sameSemantics(other._jdf)
  File "/Users/liang.zhang/mypy3/lib/python3.7/site-packages/py4j/java_gateway.py", line 1257, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/Users/liang.zhang/work/repos/apache/spark/python/pyspark/sql/utils.py", line 98, in deco
    return f(*a, **kw)
  File "/Users/liang.zhang/mypy3/lib/python3.7/site-packages/py4j/protocol.py", line 332, in get_return_value
    format(target_id, ".", name, value))
py4j.protocol.Py4JError: An error occurred while calling o42.sameSemantics. Trace:
py4j.Py4JException: Method sameSemantics([class org.apache.spark.sql.Dataset]) does not exist
	at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:318)
	at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:326)
	at py4j.Gateway.invoke(Gateway.java:274)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
@WeichenXu123 is correct. It should be `self._jdf.sameSemantics(other._jdf)`.

Reply:
emm,.. it should work. Do you correctly build and test locally ?

Reply:
I could run the scala test "sameSemantics and semanticHash work" in sbt and it passed. I uninstalled & reinstalled the pyspark by `python setup.py install` in the virtualenv. Is my routine correct?